doc/source/reference/c-api/array.rst

```diff
@@ -23,8 +23,8 @@ Array structure and data access
 These macros access the :c:type:`PyArrayObject` structure members and are
 defined in ``ndarraytypes.h``. The input argument, *arr*, can be any
 :c:type:`PyObject *<PyObject>` that is directly interpretable as a
-:c:type:`PyArrayObject *` (any instance of the :c:data:`PyArray_Type`
-and itssub-types).
+:c:type:`PyArrayObject *<PyArrayObject>` (any instance of the
```

Comment:
This also seems to work:
```suggestion
:c:expr:`PyArrayObject *` (any instance of the
```

Reply:
Does `:expr` produce a link on just the `PyArrayObject` part of the text? If so, I think I'd prefer we use `expr` where possible. Do you want to do that in this PR, or leave it for a follow-up?

Reply:
Yes, `:expr:` role works as so. I'll make this change on another PR.